jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ingresscontroller/KongIngressControllerDetector.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.config.service.ingresscontroller;
+
+import io.fabric8.kubernetes.api.model.authorization.v1.SelfSubjectAccessReview;
+import io.fabric8.kubernetes.api.model.authorization.v1.SelfSubjectAccessReviewBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.eclipse.jkube.kit.common.util.KubernetesHelper;
+
+public class KongIngressControllerDetector implements IngressControllerDetector {
+  private final KubernetesClient client;
+  private static final String INGRESS_KONG_NAMESPACE = "kong";
+
+  public KongIngressControllerDetector(KubernetesClient client) {
+    this.client = client;
+  }
+
+  @Override
+  public boolean hasIngressClass() {
+    return client.network().v1().ingressClasses()
+        .list()
+        .getItems()
+        .stream()
+        .anyMatch(i -> i.getSpec().getController().equals("ingress-controllers.konghq.com/kong"));
+  }
+
+  @Override
+  public boolean isIngressControllerReady() {
+    return checkHelmInstallationAvailable() || checkManualInstallationAvailable();
+  }
+
+  @Override
+  public boolean hasPermissions() {
+    SelfSubjectAccessReview accessReview = client.authorization().v1().selfSubjectAccessReview().create(new SelfSubjectAccessReviewBuilder()
+        .withNewSpec()
+        .withNewResourceAttributes()
+        .withResource("pods")
```

Comment:
Why are we checking for pod permissions only?
We're checking ingressClasses too